packages/dd-trace/src/payload-tagging/jsonpath-plus.js

```diff
@@ -0,0 +1,2061 @@
+'use strict';
+
+// NOTE(bengl): This file is taken directly from jsonpath-plus@10.0.0
+//
+// https://github.com/JSONPath-Plus/JSONPath/blob/a04dcbac760fed48760b09f387874a36f289c3f3/dist/index-node-cjs.index-node-cjs
+//
+// The only changes are:
+// - Replace Object.hasOwn with polyfill
```

Comment:
I would either add a short description of the reason and link to this PR or mention that we can revert this inlining once we drop support for Node.js 16

Reply:
I also realized that vendoring is distribution, and distribution requires the inclusion of the original license (as the license dictates), so I've included that, and a bunch more detail, in the latest rev.